ServiceBusMessageForwarder/ServiceBusMessageForwarder.cs

```diff
@@ -0,0 +1,237 @@
+        {
+            _activityLogger = activityLogger;
+            _messageLogger = messageLogger;
+
+            _subscriptionMessageForwarder = new SubscriptionMessageForwarder(_activityLogger, _messageLogger, messagesToHandle);
+            _queueMessageForwarder = new QueueMessageForwarder(_activityLogger, _messageLogger, messagesToHandle);
+
+            _sourceConnectionString = sourceConnectionString;
+            _destinationConnectionString = destinationConnectionString;
+
+            _sourceNamespaceManager = NamespaceManager.CreateFromConnectionString(_sourceConnectionString);
+            _destinationNamespaceManager = NamespaceManager.CreateFromConnectionString(_destinationConnectionString);
+
+            _ignoreQueuesPatterns = ignoreQueues.Split(',').Where(_ => !string.IsNullOrWhiteSpace(_)).ToArray();
+            _ignoreTopicPatterns = ignoreTopics.Split(',').Where(_ => !string.IsNullOrWhiteSpace(_)).ToArray();
+            _ignoreSubscriptionsPatterns = ignoreSubscriptions.Split(',').Where(_ => !string.IsNullOrWhiteSpace(_)).ToArray();
+        }
+        
+        public void Run()
+        {
+            try
+            {
+                _activityLogger.Log("Running");
+
+                ProcessQueues();
+
+                ProcessTopics();
+
+                _activityLogger.Log("Finished running");
+            }
+            catch (Exception e)
+            {
+                _activityLogger.Log($"! Exception: {e.Message}\n\n", 0, 2);
+            }
+        }
+
+        private void ProcessQueues()
+        {
+            var queues = _sourceNamespaceManager.GetQueues();
+            var destinationQueues = _destinationNamespaceManager.GetQueues().Select(queue => queue.Path);
+            
+            _activityLogger.Log($"{queues.Count()} queue(s) found");
+
+            foreach (var queue in queues)
+            {
+                if (IsQueueIgnored(queue.Path))
+                    _activityLogger.Log($"Ignoring queue: [{queue}]");
+
+                else if (!destinationQueues.Contains(queue.Path))
+                    _activityLogger.Log($"Skipping queue, which does not exist in destination: [{queue}]");
+
+                else
+                {
+                    QueueClient sourceClient = null;
+                    QueueClient destinationClient = null;
+
+                    try
+                    {
+                        sourceClient = QueueClient.CreateFromConnectionString(_sourceConnectionString, queue.Path);
+                        destinationClient = QueueClient.CreateFromConnectionString(_destinationConnectionString, queue.Path);
+
+                        if (queue.RequiresSession)
+                            _queueMessageForwarder.ProcessSessionQueue(sourceClient, destinationClient);
+                        else
+                            _queueMessageForwarder.ProcessQueue(sourceClient, destinationClient);
+                    }
```

Comment:
Could we add a log about which queue did we process here?

Reply:
Logs are made within the method which is called here :)

Reply:
Oh I see :)